File: Problems 3100-3199/3105. Longest Strictly Increasing or Strictly Decreasing Subarray/solution.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        
        int curr = 1, res = 1;
        bool isIncreasing = false;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] > nums[i-1]) {
                curr = isIncreasing ? curr + 1 : 2;
                isIncreasing = true;
            } else if (nums[i] < nums[i-1]) {
                curr = isIncreasing ? 2 : curr + 1;
                isIncreasing = false;
            } else {
                curr = 1;
            }
            res = max(res, curr);
        }
        return res;
    }
};

File: Problems 3100-3199/3105. Longest Strictly Increasing or Strictly Decreasing Subarray/solution.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 0: return 0
        inc = dec = 1
        maxLen = 1
        for i in range(1, n):
            if nums[i] > nums[i-1]:
                inc += 1
                dec = 1
            elif nums[i] < nums[i-1]:
                dec += 1
                inc = 1
            else:
                inc = dec = 1
            maxLen = max(maxLen, inc, dec)
        return maxLen